src/dawn/IIR/ASTConverter.cpp

```diff
@@ -0,0 +1,98 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
```

Comment:
Is this the only place where we will ever do a conversion from SIR->IIR? If not I would make this class trivial and delegate to a `iir::XXXStmt convert(sir::XXXStmt)` function.

Reply:
Yes this is the only place. I have implemented your suggestion to wrap the usage of `ASTConverter` into a function which I put in `OptimizerContext.cpp` (the only place where `ASTConverter`) is used.